src/app/project/project-documents/application-sort/application-sort.component.ts

```diff
@@ -0,0 +1,243 @@
+import { User } from 'app/models/user';
+import { Constants } from 'app/shared/utils/constants';
+import { ConfigService } from 'app/services/config.service';
+
+import 'rxjs/add/operator/switchMap';
+
+
+@Component({
+  selector: 'app-application-sort',
+  templateUrl: './application-sort.component.html',
+  styleUrls: ['./application-sort.component.scss']
+})
+export class DocumentApplicationSortComponent implements OnInit, OnDestroy {
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+  // public document: Document = null;
+  public currentProject: Project = null;
+  // public publishText: string;
+  // formatBytes: (bytes: any, decimals?: number) => string;
+  public loading = true;
+  public documents: User[] = null;
+  public readonly constants = Constants;
+
+  public tableParams: TableParamsObject = new TableParamsObject();
+  public tableData: TableObject;
+  public tableColumns: any[] = [
+    {
+      name: 'Order',
+      value: 'sortOrder',
+      width: '10%',
+    },
+    {
+      name: 'Name',
+      value: 'name',
+      width: '30%'
+    },
+    {
+      name: 'Status',
+      value: 'status',
+      width: '5%'
+    },
+    {
+      name: 'Date',
+      value: 'datePosted',
+      width: '15%'
+    },
+    {
+      name: 'Type',
+      value: 'type',
+      width: '15%'
+    },
+    {
+      name: 'Milestone',
+      value: 'milestone',
+      width: '15%'
+    },
+    {
+      name: 'Legislation',
+      value: 'legislation',
+      width: '10%'
+    }
+  ];
+
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    public api: ApiService,
+    private _changeDetectionRef: ChangeDetectorRef,
+    private storageService: StorageService,
+    private searchService: SearchService,
+    private snackBar: MatSnackBar,
+    private utils: Utils,
+    private documentService: DocumentService,
+    private tableTemplateUtils: TableTemplateUtils,
+    private configService: ConfigService
+  ) {
+  }
+
+  ngOnInit() {
+    this.currentProject = this.storageService.state.currentProject.data;
+    this.storageService.state.editedDocs = [];
+
+    this.route.params
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe(params => {
+        this.tableParams = this.tableTemplateUtils.getParamsFromUrl(params, null, 10);
+        if (this.tableParams.sortBy === '') {
+          this.tableParams.sortBy = '+sortOrder,-datePosted,+displayName';
+          this.tableTemplateUtils.updateUrl(this.tableParams.sortBy, this.tableParams.currentPage, this.tableParams.pageSize, null, this.tableParams.keywords);
+        }
+        this._changeDetectionRef.detectChanges();
+      });
+
+    this.route.data
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((res: any) => {
+        if (res) {
+          if (res.documents && res.documents[0].data.meta && res.documents[0].data.meta.length > 0) {
+            this.tableParams.totalListItems = res.documents[0].data.meta[0].searchResultsTotal;
```

Comment:
It is possible to run into a race condition here. `this.tableParams` is set in line 100 but this line could possibly execute first which would then get over written when line 100 executes. This is because line 98 and line 109 return observables which are asynchronous. I ran into this in another component recently. 

If you have a second observable the relies on the result of the first, then it is best to use `switchMap` on the first and then `subscribe` to the second.